modules/geolocation/geolocation.py

```diff
@@ -129,3 +129,38 @@ def calculate_pixel_to_geo_mapping(self):
 
         # Return matrix product of mappedGeoMatrix and sourcePixelMatrixInverse
         return (mappedGeoMatrix.dot(sourcePixelMatrixInverse))
+
+    def map_location_from_pixel(self, pixels, coordinates):
+        """
+        Maps Geographical Location Coordinates in the destination image
+        
+        Parameters
+        -------
+            pixels : np.array(shape=(3,3))
+            coordinates : np.array(shape=(5,2))
+
+        Returns
+        -------
+            np.array(shape=(5,2))
+        """
+        dehomogenized = np.array([])
+        
+        # Cycle through all 2D coordinates
+        for c in coordinates:
```

Comment:
My thought, rather than do a matrix multiplication for each vector individually, is to convert the input into a single 3xm matrix through [slicing](https://www.pythoninformer.com/python-libraries/numpy/index-and-slice/), the `.T` transpose function, and then extend the row and fill it with 1s. Once that is done, multiply this matrix by the transform.

On second thought, I do not know of any function that would let you dehomogenize the resulting matrix, and the output must be formatted back. The current for loop code may also be easier to read and understand.

Reply:
Will get on this tomorrow. But for now, I have added `pixels = np.insert(pixels, 2, 1, axis = 1)` to convert all 2D coordinates of pixels to 3D coordinates with z = 1. 

However, the for loop still exists, and I will try and find a fix for that tomorrow hopefully.